### ISSUE TITLE
Error: DB connection

### ISSUE THREAD

**Issue by khalidMhd** [hgourvest/node-firebird#318]
I am facing these two error
Error: Incompatible wire encryption levels requested on client and server    at doCallback
Error2: Incompatible wire encryption levels requested on client and server

+++++++++++++ code review ++++++++++++++

const firebird = require('node-firebird');

const options = {
    host: '127.0.0.1',
    port: 3050,
    database: 'C:/Users/khalid.d/Documents/MYDB.fdb',
    user: 'SYSDBA',
    password: 'root',
    lowercase_keys: false,
    pageSize: 4096,
    role: null,
    // charset: 'WIN1253',
    // encrypt: 'AES256',
    // algorithm: 'AES256',
  };
  

exports.executeQuery = async (query, params = []) => {
  return new Promise((resolve, reject) => {
    firebird.attach(options, (err, db) => {
      if (err) {
        console.error('Error connecting to the Firebird database:', err);
        reject(err);
        return;
      }
      console.log('Connected to the Firebird database.');

      db.query(query, params, (err, result) => {
        if (err) {
          console.error('Error executing query:', err);
          reject(err);
        } else {
          console.log('Query executed successfully.');
          resolve(result);
        }

        db.detach();
      });
    });
  });
};

**Comment by maratth** [comment]
Hi,

It's because your server isn't configure to accept connection without WireCrypt because node-firebird son't support this.

See [this section](https://github.com/hgourvest/node-firebird#firebird-30-support) in readme to configure your `firebird.conf`.

I think actually node-firebird support Srp authentification also you can try to only configure `WyreCrypt = Enabled`.

**Comment by mariuz** [comment]
Same root cause and resolution as #317 (see the close-out there): the current 2.x driver ships full wire encryption and this error now only indicates a client/server `WireCrypt` configuration mismatch or Legacy_Auth against a `WireCrypt = Required` server. Closing — please upgrade to 2.x and reopen with both settings if it persists.
